pkg/healthcheck/healthcheck.go

```diff
@@ -1361,6 +1374,32 @@ func (hc *HealthChecker) checkCanGet(namespace, group, version, resource string)
 	return hc.checkCanPerformAction("get", namespace, group, version, resource)
 }
 
+func (hc *HealthChecker) checkAPIService(serviceName string) error {
+	apiServiceClient, err := apiregistrationv1client.NewForConfig(hc.kubeAPI.Config)
+	if err != nil {
+		return err
+	}
+
+	apiStatus, err := apiServiceClient.APIServices().Get(serviceName, metav1.GetOptions{})
+	if err != nil && len(apiStatus.Status.Conditions) == 0 {
+		return err
+	}
+
+	available := false
+	var errorMessage string
+	for _, condition := range apiStatus.Status.Conditions {
+		if condition.Type == "Available" && condition.Status == "True" {
+			available = true
+			break
+		}
+		errorMessage = fmt.Sprintf("%s: %s", condition.Reason, condition.Message)
+	}
+	if !available {
+		return fmt.Errorf(errorMessage)
```

Comment:
By the looks of it we only care about the Available condition indeed. If that is the case, I think we should be looking only for it. It might be the case that this is the only condition that is in there alltogether but we can make our intent clearer by doing something like: 

```
func ensureAvailable(service *v1.APIService) error {
	for _, condition := range service.Status.Conditions {
		if condition.Type == "Available" {
			if condition.Status == "True" {
				return nil
			}
			return fmt.Errorf("%s: %s", condition.Reason, condition.Message)
		}
	}
	return fmt.Errorf("cannot determine availability of %s", service.Name)
}
```

@droidnoob @Pothulapati what do you think?

Reply:
Makes total sense :)

Reply:
👍

Reply:
@droidnoob you might also want to rebase against master and resolve any conflicts before going further. Almost there :)

Reply:
Looking at this again, wouldn't this return an error if `condition.Status` is not `True` without checking all the conditions?

Reply:
Well, yes but arent we interested only in the `Available` condition anyway? We only care whether its status is True. Correct me if I am wrong

Reply:
There might be multiple `Available` conditions. So we need to check all the conditions till the status is `True`

Reply:
Ahh, alright. If that is indeed the case (and it seems so) feel free to ignore my comment.